src/BuildScriptGenerator/SemanticVersionResolver.cs

```diff
@@ -6,17 +6,33 @@
 namespace Microsoft.Oryx.BuildScriptGenerator
 {
     using System.Collections.Generic;
+    using System.Linq;
     using SemVer;
     using Version = System.Version;
 
     internal static class SemanticVersionResolver
     {
-        public static string GetMaxSatisfyingVersion(string providedVersion, IEnumerable<string> supportedVersions)
+        /// <summary>
+        /// Gets the latest version from an enumeration that is less than or equal to the provided version.
+        /// </summary>
+        /// <param name="rangeSpec">The provided range specification to match against. For example, "3.9.0", ">=3.9", "~3.0.0".</param>
+        /// <param name="supportedVersions">The enumeration of versions to search through.</param>
+        /// <param name="loose">Determines whether or not a looser RegEx is used when searching the versions.</param>
+        /// <returns>The latest version less than or equal to the provided version.</returns>
+        public static string GetMaxSatisfyingVersion(string rangeSpec, IEnumerable<string> supportedVersions, bool loose = false)
```

Comment:
May we add a test for verifying the output when the `loose` parameter is passed?